### PR TITLE
Rename c-dependencies to runtime

### PR DESCRIPTION
**Pull request by elliottt**
Rename the top-level `c-dependencies` directory to `runtime`, and update all references. This is mostly to improve discoverability of the sources, if someone is casually browing the repository contents.

### CODE REVIEW

**Review by guybedford — APPROVED** [review]
Definitely much nicer. We could even consider fully flattening this folder into the base of the repo.

**Comment by elliottt** [comment]
It's really not obvious to me why the dependency review is failing. There are not changes to Cargo.lock for either `rust-url` or `rust-encoding`, so this should behave the same as it does on main.

**Comment by guybedford** [comment]
Perhaps this is a conflict with the `runtime` name and the "fail on scopes" in https://github.com/fastly/js-compute-runtime/blob/main/.github/workflows/dependencies.yml#LL15C9-L15C32 ?

File: ci/clang-format.ignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
-c-dependencies/js-compute-runtime/third_party/wizer.h
-c-dependencies/js-compute-runtime/rust-url/rust-url.h
+runtime/js-compute-runtime/third_party/wizer.h
+runtime/js-compute-runtime/rust-url/rust-url.h
+runtime/js-compute-runtime/rust-encoding/rust-encoding.h
```
Comment (JakeChampion) [inline ci/clang-format.ignore:3]: thank you for adding this 🙏

**Pull request merged**